node/src/database/rocksdb.rs

```diff
@@ -600,6 +644,57 @@ fn deserialize_fee_key<R: Read>(r: &mut R) -> Result<(u64, [u8; 32])> {
     Ok((fee, hash))
 }
 
+struct HeaderRecord {
+    header: ledger::Header,
+    transactions_ids: Vec<[u8; 32]>,
+}
+
+impl Serializable for HeaderRecord {
+    fn write<W: Write>(&self, w: &mut W) -> io::Result<()> {
+        // Write block header
+        self.header.write(w)?;
+
+        // Write transactions count
+        let len = self.transactions_ids.len() as u32;
+        w.write_all(&len.to_le_bytes())?;
+
+        // Write transactions hashes
+        for tx_id in &self.transactions_ids {
+            w.write_all(tx_id)?;
+        }
+
+        Ok(())
+    }
+
+    fn read<R: Read>(r: &mut R) -> io::Result<Self>
+    where
+        Self: Sized,
+    {
+        // Read block header
+        let header = ledger::Header::read(r)?;
+
+        // Read transactions count
+        let mut buf = [0u8; 4];
+        r.read_exact(&mut buf)?;
+
+        let len = u32::from_le_bytes(buf);
+
+        // Read transactions hashes
+        let mut transactions_ids = vec![];
+        for pos in 0..len {
+            let mut tx_id = [0u8; 32];
```

Comment:
1. *minor*: What if we create a `[0u8; 32*len]` outside the loop and do the read operation once? This might also allow this code to vectorise for imense performance benefits (unless this compiles to wasm)

Reply:
Not needed.